### PR TITLE
node: update solana watcher to use new solana-go version

### PR DESCRIPTION
**Pull request by bruce-riley**
The current solana watcher is failing with errors like:

`ERROR    root.solwatch-finalized    failed to request block    {"error": "-32015:Transaction version (0) is not supported by the requesting client. Please try the request again with the following configuration parameter: \"maxSupportedTransactionVersion\": 0", "slot": 155822815, "commitment": "finalized"}`

The current version of solana-go does not support this new parameter, so we need to upgrade.

### CODE REVIEW

**Review by jynnantonix — APPROVED** [review]

File: node/pkg/watchers/solana/client.go [hunk 1]
```diff
@@ -304,10 +306,15 @@ func (s *SolanaWatcher) fetchBlock(ctx context.Context, logger *zap.Logger, slot
 		zap.String("commitment", string(s.commitment)))
 
 OUTER:
-	for _, tx := range out.Transactions {
-		signature := tx.Transaction.Signatures[0]
+	for _, txRpc := range out.Transactions {
+		tx, err := txRpc.GetTransaction()
+		if err != nil {
+			logger.Error("failed to unmarshal transation", zap.Error(err))
```
Comment (hendrikhofstadt) [inline node/pkg/watchers/solana/client.go:312]: typo

**Review by hendrikhofstadt — APPROVED** [review]

File: node/hack/repair_solana/repair.go [hunk 2]
```diff
@@ -262,15 +263,20 @@ func fetchTxSeq(ctx context.Context, c *rpc.Client, sig solana.Signature) (*rpc.
...
+func process(txRpc *rpc.TransactionWithMeta) (*solana.PublicKey, error) {
 	program, err := solana.PublicKeyFromBase58(*solanaAddr)
 	if err != nil {
 		log.Fatalf("Invalid program address: %v", err)
 	}
 
-	signature := tx.Transaction.Signatures[0]
+	tx, err := txRpc.GetTransaction()
+	if err != nil {
+		log.Fatalf("Failed to unmarshal transation: %v", err)
```
Comment (hendrikhofstadt) [inline node/hack/repair_solana/repair.go:274]: let's pass in the logger here and error

Comment (hendrikhofstadt) [reply node/hack/repair_solana/repair.go:274]: we still need to return the error as the logger will not exit the program

File: node/pkg/watchers/solana/client.go (hunk 1 above)
Comment (bruce-riley) [reply node/pkg/watchers/solana/client.go]: Fixed

**Review by hendrikhofstadt — APPROVED** [review]

**Review by evan-gray — APPROVED** [review]
Approving, looks like only the typo fix since last change. Rough upgrade 😬

**Comment by evan-gray** [comment]
Looking good, just waiting to test in testnet before merging.

**Review by jynnantonix — APPROVED** [review]

**Pull request merged**